avogadro/qtplugins/closecontacts/closecontacts.cpp

```diff
@@ -0,0 +1,124 @@
+/******************************************************************************
+  This source file is part of the Avogadro project.
+  This source code is released under the 3-Clause BSD License, (see "LICENSE").
+******************************************************************************/
+
+#include "closecontacts.h"
+
+#include <avogadro/core/array.h>
+#include <avogadro/core/bond.h>
+#include <avogadro/core/elements.h>
+#include <avogadro/core/neighborperceiver.h>
+#include <avogadro/qtgui/molecule.h>
+#include <avogadro/rendering/cylindergeometry.h>
+#include <avogadro/rendering/geometrynode.h>
+#include <avogadro/rendering/groupnode.h>
+
+#include <QtCore/QSettings>
+#include <QtWidgets/QDoubleSpinBox>
+#include <QtWidgets/QFormLayout>
+#include <QtWidgets/QVBoxLayout>
+#include <QtWidgets/QWidget>
+
+namespace Avogadro {
+namespace QtPlugins {
+
+using Core::Array;
+using Core::Bond;
+using Core::NeighborPerceiver;
+using QtGui::Molecule;
+using QtGui::PluginLayerManager;
+using Rendering::CylinderGeometry;
+using Rendering::GeometryNode;
+using Rendering::GroupNode;
+
+CloseContacts::CloseContacts(QObject* p) : ScenePlugin(p)
+{
+  m_layerManager = PluginLayerManager(m_name);
+  
+  QSettings settings;
+  m_maximumDistance = settings.value("closeContacts/maximumDistance", 2.5).toDouble();
+}
+
+CloseContacts::~CloseContacts() {}
+
+void CloseContacts::process(const Molecule& molecule, Rendering::GroupNode& node)
+{
+  float radius(0.1f);
+  Vector3ub color(128, 255, 64);
+
+  NeighborPerceiver perceiver(molecule.atomPositions3d(), m_maximumDistance);
+
+  GeometryNode* geometry = new GeometryNode;
+  node.addChild(geometry);
+  CylinderGeometry *cylinders = new CylinderGeometry;
```

Comment:
Personally, I'd rather the wireframe / lines, like `wireframe.cpp`:
```
    LineStripGeometry* lines = new LineStripGeometry;
    geometry->addDrawable(lines);

    lines->setColor(m_color);
    float width = m_lineWidth;
    // etc.
    lines->addLineStrip(points, colors, width);
```

Reply:
IMHO the dashed lines with configurable colors and/or width would look good. But then again, that's because it's what we did in Avo1.

Reply:
I agree. Something that stands out from regular bonds as much as possible is the goal.